dipy/workflows/vis_registeration.py

```diff
@@ -0,0 +1,278 @@
+
+        num_slices = 0
+
+        if sli_type == 'saggital':
+            num_slices = x
+            slice_type = 0
+        elif sli_type == 'coronal':
+            num_slices = y
+            slice_type = 1
+        elif sli_type == 'axial':
+            num_slices = z
+            slice_type = 2
+
+        # Creating the renderer object and setting the background.
+        renderer = window.renderer((0.5, 0.5, 0.5))
+        slices = []
+
+        for i in range(num_slices):
+            temp_slice = overlay_slices(overlay[..., 0], overlay[..., 1],
+                                        slice_type=slice_type,
+                                        slice_index=i, ret_slice=True)
+            temp_slice = temp_slice[..., None]
+            temp_slice = np.round(temp_slice, 0).astype('uint8')
+            temp_slice = np.rollaxis(temp_slice, 2, 4)
+            slice_actor = actor.slicer(temp_slice, affine, value_range)
+            renderer.add(slice_actor)
+            renderer.reset_camera()
+            renderer.zoom(1.6)
+            snap = snapshot(renderer)
+            snap1 = self.adjust_color_range(snap)
+            slices.append(snap1)
+
+        # Writing the GIF below
+        write_gif(slices, fname, fps=10)
+
+    @staticmethod
+    def check_dimensions(static, moving):
+
+        """Check the dimensions of the input images."""
+
+        if len(static.shape) != len(moving.shape):
+            raise ValueError('Dimension mismatch: The'
+                             ' input images must have same number of '
+                             'dimensions.')
+
+    def run(self, static_img_file, moving_img_file, affine_matrix_file,
+            show_mosaic=False, anim_slice_type=None, out_dir='',
+            mosaic_file='mosaic.png', animate_file='animation.gif'):
+        """
+        Parameters
+        ----------
+        static_img_file : string
+            Path to the static image file.
+
+        moving_img_file : string
+            Path to the moving image file.
+
+        affine_matrix_file : string
+            The text file containing the affine matrix.
+
+        show_mosaic : bool, optional
+            If enabled, a mosaic of the all the slices from the
+            axial plane is saved in an image (default False).
+
+        anim_slice_type : str, optional
+            A GIF animation showing the overlap of slices
+            from static and moving images will be saved.
+            The valid slice type(s): saggital, coronal, axial, None(default).
+
+        out_dir : string, optional
+            Directory to save the results (default '').
+
+        mosaic_file : string, optional
+            Name of the file to save the mosaic (default 'mosaic.png').
+
+        animate_file : string, optional
+            Name of the GIF file for saving the animation
+            (default animation.gif).
+
+        """
+
+        io_it = self.get_io_iterator()
+
+        for static_img, mov_img, affine_matrix_file in io_it:
```

Comment:
Any reason to create a var with `io_it`? It doesn't help comprehension. You can simply use it directly in the for loop.